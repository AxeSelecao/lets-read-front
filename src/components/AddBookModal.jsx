import { useState } from "react";
import { useForm } from "react-hook-form";
import {
  useGetBooksQuery,
  useAddBookMutation,
} from "../services/redux/API/booksAPI";

export const AddBookModal = ({ active, setActive }) => {
  const {
    register,
    handleSubmit,
    resetField,
    formState: { errors },
  } = useForm();

  const [isChecked1, setChecked1] = useState();

  const handleChange1 = (e) => {
    setChecked1(e.target.checked);
  };

  const [addBook, isError] = useAddBookMutation();

  const handleAddBook = async (data) => {
    await addBook(data).unwrap();
  };

  return (
    <div
      className={active ? "modal active form-wrapper" : "modal form-wrapper"}
      onClick={() => setActive(false)}
    >
      <div className="modal-content" onClick={(e) => e.stopPropagation()}>
        <form
          onSubmit={handleSubmit((data) => {
            console.log(data);

            resetField("title");
            resetField("author");
            resetField("surname");
            resetField("name");
            resetField("patronymic");
            resetField("deadline");
            handleAddBook(data);
          })}
        >
          <div className="modal-header-container">
            <img
              className="modal-icon"
              src={require("../assets/icons/book.png")}
            />
            <h2>Добавить книгу</h2>
          </div>

          <div className="cont">
            <span>Название:</span>
            <input
              type="text"
              {...register("title", {
                required: "Enter a title of the book",
              })}
            />
          </div>
          <div className="cont">
            <span>Автор:</span>
            <input
              type="text"
              {...register("author", {
                required: "Please enter name of book",
              })}
            />
          </div>
          <div className="cont">
            <span>Это книга друга?</span>
            <input
              className="checkbox"
              type="checkbox"
              {...register("friendsbook")}
              onChange={handleChange1}
            />
          </div>
          {isChecked1 == true ? (
            <div>
              <div className="cont">
                <span>Фамилия:</span>
                <input
                  type="text"
                  {...register("surname", {
                    required: "Please enter surname of a friend",
                  })}
                />
              </div>
              <div className="cont">
                <span>Имя:</span>
                <input
                  type="text"
                  {...register("name", {
                    required: "Please enter name of a friend",
                  })}
                />
              </div>
              <div className="cont">
                <span>Отчество:</span>
                <input
                  type="text"
                  {...register("patronymic", {
                    required: "Please enter patronymic of a friend",
                  })}
                />
              </div>
              <div className="cont">
                <span>Дата возврата:</span>
                <input
                  className="deadline"
                  {...register("deadline", {
                    required: true,
                  })}
                  type={"date"}
                />
              </div>
            </div>
          ) : (
            //<div>
            //  <p>Who gave it to me?</p>
            //  <select
            //    {...register("fromfriend", {
            //      required: true,
            //    })}
            //  >
            //    {friendsArr.map((friend) => {
            //      return <option value={friend}>{friend}</option>;
            //    })}
            //  </select>
            //  <p>Deadline return:</p>
            //
            //</div>
            <></>
          )}
          <div className="cont">
            <button onClick={() => setActive(false)}>Отмена</button>
            <button type="submit">Добавить</button>
          </div>
        </form>
      </div>
    </div>
  );
};
