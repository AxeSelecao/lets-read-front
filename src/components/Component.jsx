export const Component = () => {
  const data = [
    {
      author: "author",
      title: "title",
      friendsbook: false,
    },
  ];
  return (
    <div>
      {data.map((book) => {
        return (
          <div>
            <h2>{book.author}</h2>
            <h2>{book.title}</h2>
          </div>
        );
      })}
    </div>
  );
};
