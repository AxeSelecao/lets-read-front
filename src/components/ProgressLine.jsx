import { Progress } from "antd";
import React from "react";
import { ProgressLabeledStyled } from "./styles";

const mockLabels = {
  middle: <em>180/200</em>,
};

export function ProgressLine(props) {
  return (
    <ProgressLabeledStyled>
      <Progress
        strokeWidth={12}
        percent={(props.progress / props.total_pages) * 100}
        showInfo={false}
      />
      <div className="label-middle">
        {props.progress}/{props.total_pages}
      </div>
    </ProgressLabeledStyled>
  );
}
