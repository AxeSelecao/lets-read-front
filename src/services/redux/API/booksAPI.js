import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";

export const booksApi = createApi({
  reducerPath: "booksApi",
  tagTypes: ["Books"],
  //  baseQuery: fetchBaseQuery({ baseUrl: "http://localhost:8000/" }),
  baseQuery: fetchBaseQuery({
    baseUrl: "https://lets-read-back.onrender.com/",
  }),
  endpoints: (build) => ({
    getBooks: build.query({
      query: () => `books`,
      providesTags: (result) =>
        result
          ? [
              ...result.map(({ id }) => ({ type: "Books", id })),
              { type: "Books", id: "LIST" },
            ]
          : [{ type: "Books", id: "LIST" }],
    }),
    addBook: build.mutation({
      query: (body) => ({
        url: "books",
        method: "POST",
        body,
      }),
      invalidatesTags: [{ type: "Books", id: "LIST" }],
    }),
    deleteBook: build.mutation({
      query: (id) => ({
        url: `books/${id}`,
        method: "DELETE",
      }),
      invalidatesTags: [{ type: "Books", id: "LIST" }],
    }),
    returnBook: build.mutation({
      query: (data) => (
        console.log(data),
        {
          url: `books/${data.id}`,
          method: "PUT",
          body: data.body,
        }
      ),
      invalidatesTags: (result, error, { id }) =>
        result
          ? [
              { type: "Books", id },
              { type: "Books", id: "LIST" },
            ]
          : [{ type: "Books", id: "LIST" }],
    }),
    transferBook: build.mutation({
      query: (data) => (
        console.log(data),
        {
          url: `books/${data.id}`,
          method: "PUT",
          body: data.body,
        }
      ),
      invalidatesTags: (result, error, { id }) =>
        result
          ? [
              { type: "Books", id },
              { type: "Books", id: "LIST" },
            ]
          : [{ type: "Books", id: "LIST" }],
    }),
    makeNote: build.mutation({
      query: (data) => (
        console.log(data),
        console.log(JSON.stringify({ not: data.body.note }), "JSON STRING"),
        {
          url: `books/notes/${data.id}`,
          method: "PUT",
          body: JSON.stringify({ not: data.body.note }),
        }
      ),
      invalidatesTags: (result, error, { id }) =>
        result
          ? [
              { type: "Books", id },
              { type: "Books", id: "LIST" },
            ]
          : [{ type: "Books", id: "LIST" }],
    }),
  }),
});

export const {
  useGetBooksQuery,
  useAddBookMutation,
  useDeleteBookMutation,
  useReturnBookMutation,
  useTransferBookMutation,
  useMakeNoteMutation,
} = booksApi;
