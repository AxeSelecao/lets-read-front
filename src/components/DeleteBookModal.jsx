import { useDeleteBookMutation } from "../services/redux/API/booksAPI";

export const DeleteBookModal = ({ active, setActive, id, title }) => {
  const [deleteBook] = useDeleteBookMutation();
  const handleDeleteBook = async (id) => {
    await deleteBook(id).unwrap();
  };
  return (
    <div
      className={active ? "modal active" : "modal"}
      onClick={() => setActive(false)}
    >
      <div
        className="modal-content delete-modal"
        onClick={(e) => e.stopPropagation()}
      >
        <h2 style={{ display: "block" }}>
          <span className="exclamation-mark">!</span>Вы точно хотите удалить
          книгу <br></br> "{title}"?
        </h2>
        <div className="button-wrapper">
          <button
            onClick={() => {
              handleDeleteBook(id);
              setActive(false);
            }}
          >
            Да
          </button>
          <button onClick={() => setActive(false)}>Нет</button>
        </div>
      </div>
    </div>
  );
};
