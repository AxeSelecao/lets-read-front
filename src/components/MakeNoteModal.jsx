import { useForm } from "react-hook-form";
import { useMakeNoteMutation } from "../services/redux/API/booksAPI";

export const MakeNoteModal = ({ active, setActive, id, title }) => {
  const {
    register,
    handleSubmit,
    resetField,
    formState: { errors },
  } = useForm();

  const [makeNote] = useMakeNoteMutation();

  const makeNoteButton = async (id, body) => {
    await makeNote({ id, body }).unwrap();
  };

  return (
    <div
      className={active ? "modal active" : "modal"}
      onClick={() => setActive(false)}
    >
      <div
        className="modal-content delete-modal note-modal"
        onClick={(e) => e.stopPropagation()}
      >
        <h2>
          <span>
            {" "}
            <img
              className="modal-icon arrow"
              style={{ transform: "rotate(0.5turn)" }}
              src={require("../assets/icons/note.png")}
            />
          </span>
          Добавить заметку к "{title}"?
        </h2>
        <form
          onSubmit={handleSubmit((data) => {
            console.log(data);
            resetField("note");
            makeNoteButton(id, data);
          })}
        >
          <textarea
            {...register("note", {
              required: "Enter a note to the book",
            })}
          />
          <div className="button-wrapper">
            <button onClick={() => setActive(false)}>Отмена</button>
            <button
              type="submit"
              onClick={() => {
                //  setActive(false);
              }}
            >
              Сохранить
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};
