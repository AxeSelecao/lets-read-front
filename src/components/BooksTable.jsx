import { useGetBooksQuery } from "../services/redux/API/booksAPI";
import { DeleteBookModal } from "./DeleteBookModal";
import { ProgressLine } from "./ProgressLine";
import { useState } from "react";
import { ShareBookModal } from "./ShareBookModal";
import { ReturnFriendsBookModal } from "./ReturnFriendsBookModal";
import { ReturnMyBookModal } from "./ReturnMyBookModal";
import { MakeNoteModal } from "./MakeNoteModal";

export const BooksTable = () => {
  const { data = {}, isLoading } = useGetBooksQuery();
  const [title, setTitle] = useState();
  const [id, setID] = useState();
  const [friendName, setFriendName] = useState();
  const [DeleteBookModalActive, setDeleteBookModalActive] = useState(false);
  const [shareBookModalActive, setShareBookModalActive] = useState(false);
  const [returnFriendsBookModal, setReturnFriendsBookModalActive] =
    useState(false);
  const [returnMyBookModalActive, setReturnMyBookModalActive] = useState(false);
  const [makeNoteModalActive, setMakeNoteModalActive] = useState(false);

  if (isLoading) {
    return <h1 style={{ textAlign: "center" }}>Loading...</h1>;
  }

  return (
    <div id="books-table">
      <div className="books-table-header">
        <div className="header-wrapper">
          <h3>#</h3>
        </div>
        <div className="header-wrapper">
          <h3>Title</h3>
        </div>
        <div className="header-wrapper">
          <h3>Author</h3>
        </div>
        <div className="header-wrapper">
          <h3>Progress</h3>
        </div>
        <div className="header-wrapper">
          <h3>Action</h3>
        </div>
      </div>
      {data != null && data.length > 0 ? (
        data.map((book, index) => {
          return (
            <div className="books-table-item" key={index}>
              <div className="header-wrapper">
                <h3>{index + 1}</h3>
              </div>
              <div className="header-wrapper">
                <h3>{book.title}</h3>
              </div>
              <div className="header-wrapper">
                <h3>{book.author}</h3>
              </div>
              <div className="header-wrapper progress">
                <ProgressLine
                  progress={book.progress}
                  total_pages={book.total_pages}
                />
              </div>
              <div className="header-wrapper">
                <img
                  className="action-icon"
                  src={require("../assets/icons/delete.png")}
                  onClick={() => {
                    setDeleteBookModalActive(true);
                    setID(book._id);
                    setTitle(book.title);
                  }}
                />
                {!book.friendsbook && book.name == "" ? (
                  <img
                    className="action-icon"
                    src={require("../assets/icons/share.png")}
                    onClick={() => {
                      setShareBookModalActive(true);
                      setID(book._id);
                      setTitle(book.title);
                    }}
                  />
                ) : (
                  <></>
                )}
                {!book.friendsbook && book.name != "" ? (
                  <img
                    className="action-icon arrow"
                    style={{ transform: "rotate(0.5turn)" }}
                    src={require("../assets/icons/arrow.png")}
                    onClick={() => {
                      setReturnMyBookModalActive(true);
                      setID(book._id);
                      setTitle(book.title);
                    }}
                  />
                ) : (
                  <></>
                )}
                {book.friendsbook && book.name != "" ? (
                  <img
                    className="action-icon arrow"
                    src={require("../assets/icons/arrow.png")}
                    onClick={() => {
                      setReturnFriendsBookModalActive(true);
                      setID(book._id);
                      setTitle(book.title);
                      let fullname =
                        book.surname + " " + book.name + " " + book.patronymic;
                      setFriendName(fullname);
                    }}
                  />
                ) : (
                  <></>
                )}
                <img
                  className="action-icon"
                  src={require("../assets/icons/note.png")}
                  onClick={() => {
                    setMakeNoteModalActive(true);
                    setID(book._id);
                    setTitle(book.title);
                  }}
                />
              </div>
            </div>
          );
        })
      ) : (
        <></>
      )}
      {/*<DeleteBookModal
        active={DeleteBookModalActive}
        setActive={setDeleteBookModalActive}
        id={id}
        title={title}
      />
      <ShareBookModal
        active={shareBookModalActive}
        setActive={setShareBookModalActive}
        id={id}
        title={title}
      />
      <ReturnFriendsBookModal
        active={returnFriendsBookModal}
        setActive={setReturnFriendsBookModalActive}
        id={id}
        title={title}
        friendName={friendName}
      />
      <ReturnMyBookModal
        active={returnMyBookModalActive}
        setActive={setReturnMyBookModalActive}
        id={id}
        title={title}
      />
      <MakeNoteModal
        active={makeNoteModalActive}
        setActive={setMakeNoteModalActive}
        id={id}
        title={title}
      />*/}
    </div>
  );
};
