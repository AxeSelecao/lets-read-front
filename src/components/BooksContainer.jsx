import { BooksTable } from "./BooksTable";

export const BooksContainer = () => {
  return (
    <div id="books-container">
      <h2 className="books-container-title">Books</h2>
      <BooksTable />
    </div>
  );
};
