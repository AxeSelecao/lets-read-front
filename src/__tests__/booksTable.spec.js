import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { useGetBooksQuery } from "../services/redux/API/booksAPI"; // RTK Query API
import { Component } from "../components/Component";
import { BooksTable } from "../components/BooksTable";

import * as rtkQuery from "../services/redux/API/booksAPI";

jest.mock("../services/redux/API/booksAPI");

const mockData = [
  {
    author: "author1",
    title: "my book1",
    friendsbook: false,
  },
  {
    author: "author2",
    title: "my book2",
    friendsbook: false,
  },
  {
    author: "author3",
    title: "my book3",
    friendsbook: false,
  },
];

describe("MyComponent", () => {
  it("should render data after API request", () => {
    jest
      .spyOn(rtkQuery, "useGetBooksQuery")
      .mockReturnValue({ data: mockData });
    const tree = render(<BooksTable />);

    // Check that loading state is not displayed
    // expect(screen.queryByText("Loading...")).toBeNull();
    expect(tree).toMatchSnapshot();
    // Check that data is displayed correctly
    // expect(screen.getByText("author")).toBeInTheDocument();
    // expect(screen.getByText("my book")).toBeInTheDocument();
  });
});
