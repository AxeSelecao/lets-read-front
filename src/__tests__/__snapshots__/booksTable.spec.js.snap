// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render data after API request 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="books-table"
      >
        <div
          class="books-table-header"
        >
          <div
            class="header-wrapper"
          >
            <h3>
              #
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              Title
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              Author
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              Progress
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              Action
            </h3>
          </div>
        </div>
        <div
          class="books-table-item"
        >
          <div
            class="header-wrapper"
          >
            <h3>
              1
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              my book1
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              author1
            </h3>
          </div>
          <div
            class="header-wrapper progress"
          >
            <div
              class="sc-bcXHqe iyTLPo"
            >
              <div
                class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default css-dev-only-do-not-override-10ed4xt"
                role="progressbar"
              >
                <div
                  class="ant-progress-outer"
                >
                  <div
                    class="ant-progress-inner"
                  >
                    <div
                      class="ant-progress-bg"
                      style="width: 0%; height: 12px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="label-middle"
              >
                /
              </div>
            </div>
          </div>
          <div
            class="header-wrapper"
          >
            <img
              class="action-icon"
              src="delete.png"
            />
            <img
              class="action-icon arrow"
              src="arrow.png"
              style="transform: rotate(0.5turn);"
            />
            <img
              class="action-icon"
              src="note.png"
            />
          </div>
        </div>
        <div
          class="books-table-item"
        >
          <div
            class="header-wrapper"
          >
            <h3>
              2
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              my book2
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              author2
            </h3>
          </div>
          <div
            class="header-wrapper progress"
          >
            <div
              class="sc-bcXHqe iyTLPo"
            >
              <div
                class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default css-dev-only-do-not-override-10ed4xt"
                role="progressbar"
              >
                <div
                  class="ant-progress-outer"
                >
                  <div
                    class="ant-progress-inner"
                  >
                    <div
                      class="ant-progress-bg"
                      style="width: 0%; height: 12px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="label-middle"
              >
                /
              </div>
            </div>
          </div>
          <div
            class="header-wrapper"
          >
            <img
              class="action-icon"
              src="delete.png"
            />
            <img
              class="action-icon arrow"
              src="arrow.png"
              style="transform: rotate(0.5turn);"
            />
            <img
              class="action-icon"
              src="note.png"
            />
          </div>
        </div>
        <div
          class="books-table-item"
        >
          <div
            class="header-wrapper"
          >
            <h3>
              3
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              my book3
            </h3>
          </div>
          <div
            class="header-wrapper"
          >
            <h3>
              author3
            </h3>
          </div>
          <div
            class="header-wrapper progress"
          >
            <div
              class="sc-bcXHqe iyTLPo"
            >
              <div
                class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default css-dev-only-do-not-override-10ed4xt"
                role="progressbar"
              >
                <div
                  class="ant-progress-outer"
                >
                  <div
                    class="ant-progress-inner"
                  >
                    <div
                      class="ant-progress-bg"
                      style="width: 0%; height: 12px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="label-middle"
              >
                /
              </div>
            </div>
          </div>
          <div
            class="header-wrapper"
          >
            <img
              class="action-icon"
              src="delete.png"
            />
            <img
              class="action-icon arrow"
              src="arrow.png"
              style="transform: rotate(0.5turn);"
            />
            <img
              class="action-icon"
              src="note.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="books-table"
    >
      <div
        class="books-table-header"
      >
        <div
          class="header-wrapper"
        >
          <h3>
            #
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            Title
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            Author
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            Progress
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            Action
          </h3>
        </div>
      </div>
      <div
        class="books-table-item"
      >
        <div
          class="header-wrapper"
        >
          <h3>
            1
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            my book1
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            author1
          </h3>
        </div>
        <div
          class="header-wrapper progress"
        >
          <div
            class="sc-bcXHqe iyTLPo"
          >
            <div
              class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default css-dev-only-do-not-override-10ed4xt"
              role="progressbar"
            >
              <div
                class="ant-progress-outer"
              >
                <div
                  class="ant-progress-inner"
                >
                  <div
                    class="ant-progress-bg"
                    style="width: 0%; height: 12px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="label-middle"
            >
              /
            </div>
          </div>
        </div>
        <div
          class="header-wrapper"
        >
          <img
            class="action-icon"
            src="delete.png"
          />
          <img
            class="action-icon arrow"
            src="arrow.png"
            style="transform: rotate(0.5turn);"
          />
          <img
            class="action-icon"
            src="note.png"
          />
        </div>
      </div>
      <div
        class="books-table-item"
      >
        <div
          class="header-wrapper"
        >
          <h3>
            2
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            my book2
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            author2
          </h3>
        </div>
        <div
          class="header-wrapper progress"
        >
          <div
            class="sc-bcXHqe iyTLPo"
          >
            <div
              class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default css-dev-only-do-not-override-10ed4xt"
              role="progressbar"
            >
              <div
                class="ant-progress-outer"
              >
                <div
                  class="ant-progress-inner"
                >
                  <div
                    class="ant-progress-bg"
                    style="width: 0%; height: 12px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="label-middle"
            >
              /
            </div>
          </div>
        </div>
        <div
          class="header-wrapper"
        >
          <img
            class="action-icon"
            src="delete.png"
          />
          <img
            class="action-icon arrow"
            src="arrow.png"
            style="transform: rotate(0.5turn);"
          />
          <img
            class="action-icon"
            src="note.png"
          />
        </div>
      </div>
      <div
        class="books-table-item"
      >
        <div
          class="header-wrapper"
        >
          <h3>
            3
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            my book3
          </h3>
        </div>
        <div
          class="header-wrapper"
        >
          <h3>
            author3
          </h3>
        </div>
        <div
          class="header-wrapper progress"
        >
          <div
            class="sc-bcXHqe iyTLPo"
          >
            <div
              class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-default css-dev-only-do-not-override-10ed4xt"
              role="progressbar"
            >
              <div
                class="ant-progress-outer"
              >
                <div
                  class="ant-progress-inner"
                >
                  <div
                    class="ant-progress-bg"
                    style="width: 0%; height: 12px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="label-middle"
            >
              /
            </div>
          </div>
        </div>
        <div
          class="header-wrapper"
        >
          <img
            class="action-icon"
            src="delete.png"
          />
          <img
            class="action-icon arrow"
            src="arrow.png"
            style="transform: rotate(0.5turn);"
          />
          <img
            class="action-icon"
            src="note.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
